File: stock-serveur-v2/src/main/java/cm/belrose/stockserveur/model/Vente.java
package cm.belrose.stockserveur.model;

import cm.belrose.stockserveur.config.audit.Auditable;
import lombok.*;
import org.hibernate.envers.NotAudited;

import javax.persistence.*;
import java.io.Serializable;
import java.time.Instant;
import java.util.List;

/**
 *
 * Le 09/11/2020
 *
 *@author  Ngnawen Samuel
 *
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "ventes")
public class Vente extends Auditable<String> implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String code;
    private Instant dateVente;
    private String commentaire;
    //Ce champs est juste mis pour simplifier les choses
    //@Column(name = "entreprise_id")
    private Long entrepriseId;
    @NotAudited
    @OneToMany(mappedBy = "vente")
    private List<LigneVente> ligneVenteList;
}

File: stock-client-v2/src/gs-api/src/models/vente.ts
/* tslint:disable */
import { LigneVente } from './ligne-vente';
export interface Vente {
  createdBy?: string;
  updatedBy?: string;
  createdDate?: string;
  updateDate?: string;
  id?: number;
  code?: string;
  dateVente?: number;
  commentaire?: string;
  entrepriseId?: number;
  ligneVenteList?: Array<LigneVente>;
}
